### ISSUE TITLE
wrongly limiting to 7 network interfaces per server

### ISSUE THREAD

**Issue by mbunkus** [gridscale/terraform-provider-gridscale#195]
In GS you can create no more than 8 network interfaces, which includes the connection to the public network. So far, so good. Unfortunately with the Terraform provider you can only create 7 network interfaces (again including the public network), otherwise the provider complains:

```
│ Error: Too many network blocks
│ 
│   on infrastructure.tf line 996, in resource "gridscale_server" "vpn-router1":
│  996:   network {
│ 
│ No more than 7 "network" blocks are allowed
```

I stumbled across this problem when trying to import an existing server into Terraform.

### How to reproduce

I'll start with this.

1. In GS create seven private networks
2. In GS create a server & connect it to the public network and all seven private networks (note that currently there's a bug in GS's interface that only allows you to attach seven networks when you _create_ the server, but you can _edit_ it afterwards & add the eighth interface then)
3. In GS look up the server's object UUID
4. Initialize a new Terraform installation with the GS provider
5. Add an empty `gridscale_server` resource to TF
6. Run `terraform import gridscale_server.<yourNameFromStep5> <serverObjectUUID>`
7. Run `terraform show -no-color` & copy the relevant parts from the `gridscale_server` stanza to your TF file (mostly the `network` blocks)
8. Copy the required attributes (`name`, `cores`, `memory`) from the `gridscale_server` stanza from step 7 to your TF file
9. Copy each `gridscale_server` → `network` stanza from step 7 to your TF file, but only keep their `object_uuid` property (see my example below)
10. Try to run any further TF command, e.g. `terraform validate` and observe the error message posted above

Alternatively simply create a new TF file with 7 `gridscale_network` resources (private ones), the `gridscale_public_network` data, and a `gridscale_server` resource using all 7 `gridscale_network` resources and the one `gridscale_public_network` data.

### `Terraform` Version

```
[0 mbunkus@chai-latte ~/terraform/bug-num-network-interfaces] terraform -v
Terraform v1.1.5
on linux_amd64
+ provider registry.terraform.io/gridscale/gridscale v1.14.1
```

### Affected Resource(s)

- `gridscale_server` → `network`

### Terraform Configuration Files

```terraform
terraform {
  required_providers {
    gridscale = {
      source = "gridscale/gridscale"
    }
  }
}

data "gridscale_public_network" "public" {
}

resource "gridscale_server" "test" {
  cores  = 1
  memory = 2
  name   = "bug-num-network-interfaces"

  network {
    # object_name  = "Public Network"
    object_uuid  = "f3dd72ce-1dd5-4c17-b918-124df9f1826a"
  }
  network {
    object_uuid  = "9bc71f1d-c2d9-4e77-9a77-818622bc608b"
  }
  network {
    object_uuid  = "8c2f8f22-a094-4f8c-8d5f-6c8f9655f791"
  }
  network {
    object_uuid  = "a04a772a-481d-4fe0-a8c2-2051abe2a939"
  }
  network {
    object_uuid  = "2036d44c-c4d5-4028-a95f-3f7f6f31e21c"
  }
  network {
    object_uuid  = "603e5764-d470-4b9f-bdf0-a5e4ca20d2a0"
  }
  network {
    object_uuid  = "d0c5993b-016a-4b9e-a4e3-0bd3823e7491"
  }
  network {
    object_uuid  = "63a2d6ae-a670-4133-a8cb-d32bc98cf207"
  }
}
```

### Debug Output

[Output](https://gist.github.com/mbunkus/406131cfc7bb018a9b4bc0045b5cbda8) of `TF_LOG=trace terraform validate -no-color` with above config

### Expected Behavior

The TF provider should work with 8 network interfaces as long as one of them is the public network.

The limitation should obviously not be "no more than 7 interfaces", but rather something along the lines of "no more than 7 private interfaces, allow eight if one of them is the public interface".

### Actual Behavior

The TF provider refuses to do anything at all if there are more than 7 network interfaces in a server stanza.

**Comment by nvthongswansea** [comment]
@mbunkus Thank you for reporting this issue. It will be fixed in the next release.

**Comment by nvthongswansea** [comment]
@mbunkus the issue should be fixed in `v1.14.2`. Could you check it, please?

**Comment by mbunkus** [comment]
I've given 1.14.2 a try & it seems to work just fine with the server I've had problems with before. Thanks!
